docs/src/create-apps/runtime-operations.md

```diff
@@ -0,0 +1,234 @@
+You can only trigger a runtime operation if you have permission to do so.
+Permissions are granted through the `role` option specified in the [runtime operation configuration](#define-a-runtime-operation).
+
+For example, to trigger the runtime operation [defined previously](#define-a-runtime-operation),
+you could use the following command:
+
+```bash
+platform p:curl /environments/{{< variable "ENVIRONMENT_ID" >}}/deployments/current/operations -X POST -d '{"operation": "clear-rebuild", "service": "app"}' -p {{< variable "PROJECT_ID" >}}
+```
+
+## Runtime operation examples
+
+### Build your app when using a static site generator
+
+During every Platform.sh deployment, a standard [`build` step](../overview/build-deploy.md#the-build) is run.
+When you use a static site generator like [Gatsby](../guides/gatsby/_index.md)
+or [Next.js](../guides/nextjs/_index.md) with [a headless backend](../guides/gatsby/headless/_index.md),
+you need to run another `build` step to get your app ready for production.
+
+This extra `build` step includes the generation of all the HTML pages your site can then serve in a static way.
+You can use a runtime operation to trigger it after the initial deployment of your app or after a redeployment.
+You can also trigger it when you need to fetch content from your backend system
+but want to avoid going through the whole Platform.sh [build and deploy processes](../overview/build-deploy.md) again.
+
+{{< codetabs >}}
++++
+title=Gatsby
++++
+
+To run the [Gatsby build](https://www.gatsbyjs.com/docs/conceptual/overview-of-the-gatsby-build-process/#understanding-gatsby-build-build-time) step,
+define a runtime operation similar to the following:
+
+```yaml {location=".platform.app.yaml"}
+operations:
+  gatsby-build:
+    role: viewer
+    commands:
+      start: gatsby build
+      stop: null
+```
+
+To trigger your runtime operation, run a cURL command similar to the following:
+
+```bash
+$ platform p:curl /environments/{{< variable "ENVIRONMENT_ID" >}}/deployments/current/operations -X POST -d '{"operation": "gatsby-build", "service": "{{< variable "CONTAINER_NAME" >}}"}' -p {{< variable "PROJECT_ID" >}}
+```
+
+<--->
++++
+title=Next.js
++++
+
+To run the [Next.js build](https://nextjs.org/docs/deployment#nextjs-build-api) step,
+define a runtime operation similar to the following:
+
+```yaml {location=".platform.app.yaml"}
+operations:
+    next-build:
+        role: admin
+        commands:
+            # All below are valid, depending on your setup
+            start: next build
+            # start: npx next build
+            # start: npm run build
+            stop: null
+```
+
+To trigger your runtime operation, run a cURL command similar to the following:
+
+```bash
+$ platform p:curl /environments/{{< variable "ENVIRONMENT_ID" >}}/deployments/current/operations -X POST -d '{"operation": "next-rebuild", "service": "{{< variable "CONTAINER_NAME" >}}"}' -p {{< variable "PROJECT_ID" >}}
+```
+
+{{< /codetabs >}}
+
```

Comment:
```suggestion

{{< note >}}

Both of the examples above assume that the frontend and backend containers are included in the same environment. 
The frontend pulls content-related data from the backend's API during that frontend framework's build.
In order to accomplish this on Platform.sh, where each application goes through a build-deploy pipeline in parallel, the frontend's build must be delayed _until after_ the backend has fully deployed. 
This will often be as late as `post_deploy` hook, when the filesystem is read-only. 

Having both frontend and backend containers in the same environment is not necessary to run this command.
What _is_ necessary is that the build destination for the frontend _must_ be writable at runtime (i.e., a mount has been defined for it). 
If including a build within a mount is undesirable - especially in those cases when the data source is _not_ on Platform.sh, look to [source operations](/create-apps/source-operations.md) instead to achieve a similar effect, but through generating a new commit. 

{{< /note  >}}

```

Reply:
Remi's feedback here is correct, and since it's not super explicit, I took a stab at a note (that's probably too long).